### PR TITLE
Tensorflow-lite custom bin support: Plan B @open sesame 6/10 08:01

### PR DESCRIPTION
**Pull request by myungjoo**
This addresses plan B of #3713
    
    $ ldd build/ext/nnstreamer/tensor_filter/libnnstreamer_filter_tensorflow2-lite-custom.so
            linux-vdso.so.1 (0x00007fffe21f2000)
            libnnstreamer-single.so => /source/AutoDrv/NNStreamer/build/ext/nnstreamer/tensor_filter/../../../gst/nnstreamer/libnnstreamer-single.so (0x00007f08b5bba000)
            /source/AutoDrv/NNStreamer/ext/nnstreamer/tensor_filter/libtensorflow2-lite-custom.so (0x00007f08b50a5000)
            libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f08b4ea1000)
            libglib-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007f08b4b8a000)
            libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f08b477d000)
            libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f08b4565000)
            libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f08b4174000)
            libgmodule-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0 (0x00007f08b3f70000)
            libgobject-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0 (0x00007f08b3d1c000)
            libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f08b3afd000)
            /lib64/ld-linux-x86-64.so.2 (0x00007f08b5ff8000)
            libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007f08b388b000)
            libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f08b34ed000)
            libffi.so.6 => /usr/lib/x86_64-linux-gnu/libffi.so.6 (0x00007f08b32e5000)
    
    After this, we need the following commits.
    1. a manual how to use tflite2-custom. (at /ext/nnstreamer/tensor-filter?)
    2. a test case for tflite2-custom.

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

### CODE REVIEW

**Comment by myungjoo** [comment]
Issue to be resolved before merge.

1. let rpm/deb installer ignore the (shlib) dependency on libtensorflow2-lite-custom.so from the package
2. make a "gentle" error with libtensorflow2-lite-custom.so not found error

File: ext/nnstreamer/tensor_filter/meson.build [hunk 1]
```diff
@@ -280,6 +283,90 @@ if tflite2_support_is_available
...
+
+    if filepath == ''
+      ## No shared tfl lib found. Get static lib and convert it
+
+      filepaths = [join_paths(nnstreamer_libdir, 'libtensorflow2-lite.a'),
+          '/usr/lib/libtensorflow2-lite.a',
+          '/usr/lib64/libtensorflow2-lite.a',
+          './libtensorflow2-lite.a',
+          join_paths(meson.current_build_dir(), 'libtensorflow2-lite.a',),
+          join_paths(meson.current_source_dir(), 'libtensorflow2-lite.a',)]
```
Comment (anyj0527) [inline ext/nnstreamer/tensor_filter/meson.build:327]: LGTM👍 Minor concern: the order of L297~L303 and L322~L327 should be same?

**Review by anyj0527 — APPROVED** [review]

File: ext/nnstreamer/tensor_filter/meson.build (hunk 1 above)
Comment (myungjoo) [reply ext/nnstreamer/tensor_filter/meson.build:327]: No. they were ordered differently intentionally.

.a is supposed to be provided by our tf-lite-dev package (the first three cases) primarily.

.so is supposed to be provided by the builder (we don't release tf-lite so in our packages) as an ad-hoc method; thus, they are supposed to be at the current "source" or "bin" directory.

File: packaging/nnstreamer.spec [hunk 2]
```diff
@@ -163,6 +163,10 @@ BuildRequires:	gst-plugins-bad-devel
 BuildRequires:	glib2-devel
 BuildRequires:	meson >= 0.50.0
 
+# tensorflow2-lite-custom requires scripts for rpm >= 4.9
+BuildRequires:  rpm >= 4.9
```
Comment (jaeyun-jung) [inline packaging/nnstreamer.spec:167]: Q: condition to check tf-lite2?
%if 0%{?tensorflow2_lite_support}
BuildRequires: rpm >= 4.9
...

Comment (jaeyun-jung) [inline packaging/nnstreamer.spec:167]: Q: condition to check tf-lite2?
%if 0%{?tensorflow2_lite_support}
BuildRequires: rpm >= 4.9
...

Comment (myungjoo) [reply packaging/nnstreamer.spec]: Moved the line into the ```%if / %endif``` line.

**Review by jaeyun-jung — APPROVED** [review]

**Pull request merged**